src/lib/containers/entity_finder/EntityFinder.tsx

```diff
@@ -172,50 +105,70 @@ export const EntityFinder: React.FunctionComponent<EntityFinderProps> = ({
     type: EntityDetailsListDataConfigurationType.PROMPT,
   })
 
-  const isSelected = (entity: Reference) => {
-    return selectedEntities.some(
+  const handleError = useErrorHandler()
+
+  const setBreadcrumbs = useCallback(
+    (items: BreadcrumbItem[]) => {
+      setBreadcrumbsProps({
+        items,
+      })
+    },
+    [setBreadcrumbsProps],
+  )
+
+  function isSelected(entity: Reference, selected: Reference[]): boolean {
+    return selected.some(
       s =>
         s.targetId === entity.targetId &&
         s.targetVersionNumber === entity.targetVersionNumber,
     )
   }
 
-  const otherVersionSelected = (entity: Reference) => {
-    return selectedEntities.some(
+  function otherVersionSelected(
+    entity: Reference,
+    selected: Reference[],
+  ): boolean {
+    return selected.some(
       s =>
         s.targetId === entity.targetId &&
         s.targetVersionNumber !== entity.targetVersionNumber,
     )
   }
 
-  const toggleSelection = (entity: Reference) => {
-    if (isSelected(entity)) {
+  function entitySelectionReducer(
```

Comment:
Much easier to follow the logic in updating the `selectedEntities` state in a reducer function